src/ucp/core/ucp_context.c

```diff
@@ -865,62 +859,103 @@ static int ucp_is_resource_in_device_list(const uct_tl_resource_desc_t *resource
 }
 
 static int
-ucp_is_resource_in_transports_list(const char *tl_name,
-                                   const ucs_config_allow_list_t *allow_list,
-                                   uint8_t *rsc_flags, uint64_t *tl_cfg_mask)
+ucp_is_tl_in_transports_list(const char *tl_name,
```

Comment:
maybe this function should return combination of 3 bits - regardless of allow_mode:
1. transport is in the list as regular transport (no aux)
2. transport is in the "main" list as aux
3. transport is in the "alias" list as aux

Then, the caller function will combine the "allow_mode" logic with this result
this way we fully separate between "understanding the configuration" and "applying allow-mode logic"

Reply:
The problem here is that there different interpretation of __transport is in the "alias" list as aux__ for allow and deny list.
E.g. `UCX_TLS=rc_x` should enable `rc_mlx5`, and `ud_mlx5` as auxiliary. `UCX_TLS=^rc_x` should only disable `rc_mlx5`, and do nothing with `ud_mlx5`. In the same time `UCX_TLS=ud_x` should disable `ud_mlx5`, but keep it available as auxiliary.
Taking into account this behavior we cannot make a simple check for three state inside the function. It can be done only using 4 states.